ceph/rados/core_workflows.py

```diff
@@ -451,35 +451,56 @@ def bench_write(self, pool_name: str, **kwargs) -> bool:
         try:
             self.node.shell([cmd], check_status=check_ec, timeout=_timeout)
             if max_objs and verify_stats:
-                time.sleep(90)
-                new_objs = self.get_cephdf_stats(pool_name=pool_name)["stats"][
-                    "objects"
-                ]
-                log.info(
-                    f"Objs in the {pool_name} before IOPS: {org_objs} "
-                    f"| Objs in the pool post IOPS: {new_objs} "
-                    f"| Expected {org_objs + max_objs} or {org_objs + max_objs + 1}"
-                )
-                assert (new_objs == org_objs + max_objs) or (
-                    new_objs == org_objs + max_objs + 1
-                )
+                exp_objs = org_objs + max_objs
+                assert self.verify_pool_stats(pool_name=pool_name, exp_objs=exp_objs)
             else:
                 time.sleep(15)
                 new_objs = self.get_cephdf_stats(pool_name=pool_name)["stats"][
                     "objects"
                 ]
-                log.info(
-                    f"Objs in the {pool_name} before IOPS: {org_objs} "
-                    f"| Objs in the pool post IOPS: {new_objs} "
-                    f"| Expected {new_objs} > 0"
-                )
+                log_info_msg = f"Objs in the {pool_name} before IOPS: {org_objs} \
+                    | Objs in the pool post IOPS: {new_objs} \
+                    | Expected {new_objs} > 0 | Expected {new_objs} > {org_objs}"
+                log.info(log_info_msg)
                 assert new_objs > 0
+                assert new_objs > org_objs
             return True
         except Exception as err:
             log.error(f"Error running rados bench write on pool : {pool_name}")
             log.error(err)
             return False
 
+    def verify_pool_stats(self, pool_name, exp_objs: int, timeout=180) -> bool:
+        """
+        Method to verify pool stats
+        Args:
+            pool_name: pool on which the operation will be performed ( str )
+            exp_objs -> expected objects on the pool (int)
+            timeout (int) -> user defined timeout for verify pool stats
+        Returns: True -> pass, False -> fail
+        """
+        end_time = datetime.datetime.now() + datetime.timedelta(seconds=timeout)
+        while end_time > datetime.datetime.now():
+            new_objs = self.get_cephdf_stats(pool_name=pool_name)["stats"]["objects"]
+            log_info_msg = f"| Objs in the pool post IOPS: {new_objs} | Expected {exp_objs} or {exp_objs + 1}"
+            log.info(log_info_msg)
+            if (new_objs == exp_objs) or (new_objs == exp_objs + 1):
+                log.info("Stats in the pool are as expected")
+                break
+
+            log_info_msg = (
+                f"Stats in the pool are yet to match expected object count {exp_objs}"
+            )
+            log.info(log_info_msg)
```

Comment:
We can make these statements debug. print only end goals in info. for all intermittent data, let's add them in debug.